x-pack/plugins/enterprise_search/server/collectors/app_search/telemetry.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { set } from 'lodash';
+import { ISavedObjectsRepository, SavedObjectsServiceStart } from 'src/core/server';
+import { UsageCollectionSetup } from 'src/plugins/usage_collection/server';
+
+import { AS_TELEMETRY_NAME, ITelemetrySavedObject } from '../../saved_objects/app_search/telemetry';
+
+/**
+ * Register the telemetry collector
+ */
+
+interface Dependencies {
+  savedObjects: SavedObjectsServiceStart;
+  usageCollection: UsageCollectionSetup;
+}
+
+export const registerTelemetryUsageCollector = ({
+  usageCollection,
+  savedObjects,
+}: Dependencies) => {
+  const upgradeAssistantUsageCollector = usageCollection.makeUsageCollector({
+    type: AS_TELEMETRY_NAME,
+    fetch: async () => fetchTelemetryMetrics(savedObjects),
+  });
+  usageCollection.registerCollector(upgradeAssistantUsageCollector);
+};
+
+/**
+ * Fetch the aggregated telemetry metrics from our saved objects
+ */
+
+const fetchTelemetryMetrics = async (savedObjects: SavedObjectsServiceStart) => {
+  const savedObjectsRepository = savedObjects.createInternalRepository();
```

Comment:
Is this method OK to call on every request? Or is it something that is meant to only be called once to create a repository.

Reply:
I actually don't know the answer to this myself, sorry, I was just grabbing whatever Upgrade Assistant was using. I assume it's OK since another plugin is already using it / it's working as-is? We can definitely flag it and ask the Kibana team during review however.